pkg/scheduler/queue/tenant_queues_test.go

```diff
@@ -916,3 +918,36 @@ func (n *Node) nodeCount() int {
 	}
 	return count
 }
+
+// getNextTenantForQuerier is a test utility, not intended for use by consumers of queueBroker
+//
+// The next tenant for the querier is obtained by rotating through the global tenant order
+// starting just after the last tenant the querier received a request for, until a tenant
+// is found that is assigned to the given querier according to the querier shuffle sharding.
+// A newly connected querier provides lastTenantIndex of -1 in order to start at the beginning.
+func (qb *queueBroker) getNextTenantForQuerier(lastTenantIndex int, querierID QuerierID) int {
```

Comment:
I have not bothered with this much myself, but maybe we can move toward just making the relevant type the first argument of a function:
`func getNextTenantForQuerier(qb *queueBroker, lastTenantIndex int, querierID QuerierID) int {`
instead of using methods and commenting them to say test utility only.